android-emulator-plugin/src/main/java/com/quittle/androidemulator/AndroidEmulatorPlugin.java

```diff
@@ -153,9 +153,14 @@ private static void createInstallEmulatorSystemImageTask(final Project project,
     private static void createCreateEmulatorTask(final Project project, final EmulatorConfiguration emulatorConfiguration) {
         final File avdRoot = emulatorConfiguration.getAvdRoot();
         final String emulatorName = emulatorConfiguration.getEmulatorName();
+        final String emulatorDevice = emulatorConfiguration.getDeviceType();
         createExecTask(project, emulatorConfiguration, CREATE_ANDROID_EMULATOR_TASK_NAME, exec -> {
                     exec.setExecutable(emulatorConfiguration.getAvdManager());
-                    exec.setArgs(l("create", "avd", "--name", emulatorName, "--package", emulatorConfiguration.getSystemImagePackageName(), "--force"));
+                    if (emulatorDevice == null) {
+                        exec.setArgs(l("create", "avd", "--name", emulatorName, "--package", emulatorConfiguration.getSystemImagePackageName(), "--force"));
+                    } else {
+                        exec.setArgs(l("create", "avd", "--name", emulatorName, "--package", emulatorConfiguration.getSystemImagePackageName(), "--device", emulatorDevice, "--force"));
+                    }
```

Comment:
Could you instead create a list before and optionally extend if it `emulatorDevice` is not null? That way these two lists can't get out of sync with future changes.